File: core/ifw-api/src/main/java/com/merxury/ifw/entity/Component.java
/*
 * Copyright 2022 Blocker
 *
 *   Licensed under the Apache License, Version 2.0 (the "License");
 *   you may not use this file except in compliance with the License.
 *   You may obtain a copy of the License at
 *
 *       https://www.apache.org/licenses/LICENSE-2.0
 *
 *   Unless required by applicable law or agreed to in writing, software
 *   distributed under the License is distributed on an "AS IS" BASIS,
 *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *   See the License for the specific language governing permissions and
 *   limitations under the License.
 */

package com.merxury.ifw.entity;

import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.Element;
import org.simpleframework.xml.ElementList;

import java.util.List;

public class Component {
    @Attribute
    protected boolean block = true;

    @Attribute
    protected boolean log = false;

    @ElementList(entry = "component-filter", inline = true, empty = false, required = false)
    protected List<ComponentFilter> componentFilters;

    @Element(name = "intent-filter", required = false)
    protected IntentFilter intentFilter;

    public List<ComponentFilter> getComponentFilters() {
        return componentFilters;
    }

    public void setComponentFilters(List<ComponentFilter> componentFilters) {
        this.componentFilters = componentFilters;
    }

    public IntentFilter getIntentFilter() {
        return intentFilter;
    }

    public void setIntentFilter(IntentFilter intentFilter) {
        this.intentFilter = intentFilter;
    }

    public boolean isBlock() {
        return block;
    }

    public void setBlock(boolean block) {
        this.block = block;
    }

    public boolean isLog() {
        return log;
    }

    public void setLog(boolean log) {
        this.log = log;
    }
}

File: core/data/src/main/java/com/merxury/blocker/core/data/model/Component.kt
/*
 * Copyright 2023 Blocker
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.merxury.blocker.core.data.model

import com.merxury.blocker.core.model.ComponentType

data class Component(
    val name: String,
    val simpleName: String,
    val packageName: String,
    val type: ComponentType,
    val pmBlocked: Boolean,
    val ifwBlocked: Boolean = false,
    val description: String? = null,
)
